lib/src/tasks/gen_test_runner/api.dart

```diff
@@ -0,0 +1,130 @@
+  currentConfig.genHtml ? args.add('--genHtml') : args.add('--no-genHtml');
+
+  GenTestRunnerTask task =
+      new GenTestRunnerTask('$taskTitle ${args.join(' ')}');
+
+  File generatedRunner =
+      new File('${currentConfig.directory}/${currentConfig.filename}.dart');
+  IOSink writer = generatedRunner.openWrite(mode: FileMode.WRITE);
+
+  Directory testDirectory = new Directory(currentConfig.directory);
+  List<File> testFiles = [];
+  List<FileSystemEntity> allFiles =
+      testDirectory.listSync(recursive: true, followLinks: false);
+  allFiles.forEach((FileSystemEntity entity) {
+    if (entity is File) {
+      if (entity.path.endsWith('_test.dart')) {
+        testFiles.add(entity);
+        task.testFiles.add(entity.path);
+      } else if (!entity.path.endsWith('${currentConfig.filename}.dart') &&
+          entity.path.endsWith('.dart')) {
+        task.excludedFiles.add(entity.path);
+      }
+    }
+  });
+
+  if (currentConfig.env == Environment.browser) {
+    if (currentConfig.genHtml) {
+      await testHtmlFileGenerator(currentConfig.directory,
+          currentConfig.filename, currentConfig.scriptPaths);
+    }
+    writer.writeln('@TestOn(\'browser\')');
+  } else {
+    writer.writeln('@TestOn(\'vm\')');
+  }
+  writer.writeln(
+      'library ${currentConfig.directory.replaceAll('/','.')}.${currentConfig.filename};');
+  writer.writeln();
+
+  testFiles.forEach((File file) {
+    Match filenameMatch = new RegExp(r'([^/]+).dart$').firstMatch(file.path);
+    String filename = filenameMatch.group(1);
+    writer.writeln(
+        'import \'${file.path.replaceFirst(currentConfig.directory, '.')}\' as ${filename};');
+  });
+
+  writer.writeln('import \'package:test/test.dart\';');
+
+  if (currentConfig.additionalImports.isNotEmpty) {
+    currentConfig.additionalImports.forEach((String import) {
+      writer.writeln(import);
+    });
+  }
+
+  writer.writeln('');
+  writer.writeln('void main() {');
+
+  if (currentConfig.commandsPriorToTesting.isNotEmpty) {
+    currentConfig.commandsPriorToTesting.forEach((String command) {
+      writer.writeln('  $command');
+    });
+  }
+
+  testFiles.forEach((File file) {
+    Match filenameMatch = new RegExp(r'([^/]+).dart$').firstMatch(file.path);
+    writer.writeln('  ${filenameMatch.group(1)}.main();');
+  });
+
+  writer.writeln('}');
+  await writer.close();
+
+  task.runnerFile = '${currentConfig.directory}/${currentConfig.filename}.dart';
+
+  task.successful = true;
+
+  return task;
+}
+
+Future testHtmlFileGenerator(
+    String directory, String filename, List<String> scriptPaths) async {
+  File generatedRunner = new File('$directory/$filename.html');
+  IOSink writer = generatedRunner.openWrite(mode: FileMode.WRITE);
+  writer.writeln('<!DOCTYPE html>');
+  writer.writeln('<html>');
+  writer.writeln('  <head>');
+  writer.writeln('    <title>$filename</title>');
+  scriptPaths.forEach((path) {
+    writer.writeln('    <script src="$path"></script>');
```

Comment:
there's a discrepancy in the usage between imports and scripts.  imports appear to expect that the user includes the full import line (`import 'package:fluri/fluri.dart';`), which seems reasonable since that will allow users to include things like `... as whatever;`, but the scripts expect only the script source path and infer the rest of the line.

what would you guys think about just using something like `dartHeaders` and `htmlHeaders` and expecting the full line in both places?  this would allow the dart headers to include things beyond imports (maybe annotation lines?) and the html headers to include things beyond scripts (maybe css?).  Can't think of a slew of use cases for this off the top of my head, but it's worth considering